crates/core/tedge_api/src/workflow/state.rs

```diff
@@ -176,39 +239,86 @@ impl GenericCommandState {
     }
 
     pub fn operation(&self) -> Option<String> {
-        match self.topic.name.split('/').collect::<Vec<&str>>()[..] {
-            [_, _, _, _, _, "cmd", operation, _] => Some(operation.to_string()),
-            _ => None,
-        }
+        extract_command_identifier(&self.topic.name).map(|(operation, _)| operation)
     }
 
     pub fn cmd_id(&self) -> Option<String> {
-        match self.topic.name.split('/').collect::<Vec<&str>>()[..] {
-            [_, _, _, _, _, "cmd", _, cmd_id] => Some(cmd_id.to_string()),
-            _ => None,
-        }
+        extract_command_identifier(&self.topic.name).map(|(_, cmd_id)| cmd_id)
     }
 
-    pub fn is_terminal(&self) -> bool {
-        matches!(self.status.as_str(), "successful" | "failed")
+    pub fn is_successful(&self) -> bool {
+        matches!(self.status.as_str(), SUCCESSFUL)
+    }
+
+    pub fn is_failed(&self) -> bool {
+        matches!(self.status.as_str(), FAILED)
+    }
```

Comment:
thought: Lots of methods attached to this type, could some potentially be extracted or placed elsewhere?

Reply:
`GenericCommandState` sounds to be a good place for these methods.